.github/workflows/amplify_canaries.yaml

```diff
@@ -156,3 +157,35 @@ jobs:
         run: |
           flutter build ios --simulator --target=integration_test/main_test.dart
           flutter test -d test integration_test/main_test.dart
+
+  log-failure-metric:
+    # Send failure data point to Cloudwatch metric CanaryTestFailure on failure
+    # See repo amplify-ci-support/github-cloudwatch-cdk for related CDK Cloudwatch setup
+    runs-on: macos-latest
+    environment: ci
+    needs: build
+    if: ${{ failure() }}
+    steps:
```

Comment:
I get this error so I think you need to call checkout here as it will not have context/env from prereq job

```
Error: Can't find 'action.yml', 'action.yaml' or 'Dockerfile' under '/Users/runner/work/amplify-flutter/amplify-flutter/.github/composite_actions/log_metric'. Did you forget to run actions/checkout before running your local action?
```

Reply:
That's strange, been checking the UI team's implementation and they don't need to do that.. Perhaps it's because they use environment: ci but I don't see us using environment in our setup.